specs/bls_signature.md

```diff
@@ -68,45 +68,30 @@ We require:
 ### `hash_to_G2`
 
 ```python
-G2_cofactor = 305502333931268344200999753193121504214466019254188142667664032982267604182971884026507427359259977847832272839041616661285803823378372096355777062779109
-q = 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559787
-
-def hash_to_G2(message_hash: Bytes32, domain: Bytes8) -> Tuple[uint384, uint384]:
-    # Initial candidate x coordinate
-    x_re = int.from_bytes(hash(message_hash + domain + b'\x01'), 'big')
-    x_im = int.from_bytes(hash(message_hash + domain + b'\x02'), 'big')
-    x_coordinate = Fq2([x_re, x_im])  # x = x_re + i * x_im
-    
-    # Test candidate y coordinates until a one is found
-    while 1:
-        y_coordinate_squared = x_coordinate ** 3 + Fq2([4, 4])  # The curve is y^2 = x^3 + 4(i + 1)
-        y_coordinate = modular_squareroot(y_coordinate_squared)
-        if y_coordinate is not None:  # Check if quadratic residue found
-            return multiply_in_G2((x_coordinate, y_coordinate), G2_cofactor)
-        x_coordinate += Fq2([1, 0])  # Add 1 and try again
+def hash_to_G2(message_hash: Bytes32) -> Tuple[uint384, uint384]:
+    hash_to_curve(message_hash)
```

Comment:
Can we just rename `hash_to_G2 ` to `hash_to_curve`?

Reply:
Oh I've been spending too much time in rust land I've forgotten my return statements. I think it should be `return hash_to_curve(message_hash)`

I was very tempted to rename `hash_to_g2` however as `hash_to_g2` takes `Bytes32` whereas `hash_to_curve` takes `Bytes` I decided to make the abstraction. What are your thoughts on this?